solutions/cf-1514C.mdx

```diff
@@ -0,0 +1,84 @@
+---
+id: cf-1514C
+source: CF
+title: Product 1 Modulo N
+author: Jesse Choe
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/89810)
+
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+using ll = long long;
+
+ll gcd(ll a, ll b){
```

Comment:
c++17 has built in gcd but this is fine too

Reply:
```suggestion
ll gcd(ll a, ll b) {
```